lib/mongoid/contextual/none.rb

```diff
@@ -6,11 +6,27 @@ module Mongoid
   module Contextual
     class None
       include Enumerable
+      alias :old_sum :sum
+
       include Aggregable::None
       include Queryable
 
       attr_reader :criteria, :klass
 
+      # Get the sum in the null context.
+      #
+      # @example Get the sum of null context.
+      #     context.sum(_field)
+      #
+      # @param [ Symbol ] _field The field to sum.
+      #
+      # @return [ Integer | String ] If the return_nil_on_aggregate_none is
+      #   turn one, this will always be zero. Otherwise, it will return the
+      #   field name as string.
+      def sum(_field = nil)
```

Comment:
Aliasing the old `sum` and using it here is only necessary for the `sum` function. Before original ticket, the null context did not implement `aggregates`, `avg`, `min`, or `max` at all. Since this is an addition I don't think it needs to be put behind a feature flag.